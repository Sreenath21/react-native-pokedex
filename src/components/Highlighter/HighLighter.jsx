import React from 'react';
import {View, Text} from 'react-native';

import {capitalize} from '../../utils/helplerFunctions';
import colors from '../../utils/colors';
import styles from './Highlighter.style';

const HighLighter = ({textArray}) => {
  return (
    <View style={styles.container}>
      {textArray?.map(text => (
        <View
          style={[
            styles.wrapper,
            {backgroundColor: colors[text.toLowerCase()]},
          ]}
          key={text}>
          <Text style={styles.text}>{capitalize(text)}</Text>
        </View>
      ))}
    </View>
  );
};

export default HighLighter;
