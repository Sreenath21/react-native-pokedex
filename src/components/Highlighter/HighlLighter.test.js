import React from 'react';
import HighLighter from './HighLighter';
import renderer from 'react-test-renderer';

it('HighLighter renders correctly', () => {
  const tree = renderer.create(<HighLighter textArray={['Normal', 'Grass']} />);
  expect(tree).toMatchSnapshot();
});
