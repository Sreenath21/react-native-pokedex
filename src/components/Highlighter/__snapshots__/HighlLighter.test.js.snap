// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HighLighter renders correctly 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "flexWrap": "wrap",
      "gap": 5,
    }
  }
>
  <View
    style={
      [
        {
          "borderColor": "#391363",
          "borderRadius": 4,
          "borderWidth": 1,
          "paddingHorizontal": 5,
          "paddingVertical": 1,
        },
        {
          "backgroundColor": "#DDCBD0",
        },
      ]
    }
  >
    <Text
      style={
        {
          "fontSize": 16,
        }
      }
    >
      Normal
    </Text>
  </View>
  <View
    style={
      [
        {
          "borderColor": "#391363",
          "borderRadius": 4,
          "borderWidth": 1,
          "paddingHorizontal": 5,
          "paddingVertical": 1,
        },
        {
          "backgroundColor": "#C0D4C8",
        },
      ]
    }
  >
    <Text
      style={
        {
          "fontSize": 16,
        }
      }
    >
      Grass
    </Text>
  </View>
</View>
`;
